### DISCUSSION TITLE
Add Functionnality

### DISCUSSION THREAD

**Discussion by jlebard** [lbr38/repomanager#137] [Ideas]
Hello, 
Sorry for my bad english :) : list of features when u can think : 
-  API : Post package into repository
- API : Rebuild Repo
- UI: Authentification : OpenID
- CORE: Proxy Cache: not a mirror, juste a proxy for packages with cache (like Apt-Cacher-NG)
Thank u for your very good job,  we beginning with your tools,, and if we continue to use, i try to tips u :) 
Bye, good days 
JL

**Comment by lbr38** [comment]
Hello @jlebard 

Thank you for your interest in the project, I really appreciate it!

You can write me in English or French if that can help you ;-)

* API : Post package into repository => **Good news there is already an endpoint to do this (it is all new!). I just didn't take the time to update the API documentation, I will do it this week-end and let you know.**
* API : Rebuild Repo => **Same as above, I'll let you know when I've updated the documentation.**
* UI: Authentification : OpenID => **Unfortunately, I do not have any plans to add new authentication features at the moment. Authentication is a very sensitive subject, sometimes complex to implement and especially to develop. Additionally, I have no testing environment to test authentication solutions.**
* CORE: Proxy Cache: not a mirror, juste a proxy for packages with cache (like Apt-Cacher-NG) => **It is not the purpose of Repomanager to serve as a proxy.**

**Comment by lbr38** [comment]
@jlebard I updated the API documentation and added the two new endpoints to:

- upload package to a repository snapshot ``https://<FQDN>/api/v2/snapshot/<SNAPSHOT_ID>/upload``
- rebuild repository snapshot metadata ``https://<FQDN>/api/v2/snapshot/<SNAPSHOT_ID>/rebuild``

API documentation : https://github.com/lbr38/repomanager/wiki/10.-API#endpoints

Tell me if you need any help.

**Comment by jlebard** [comment]
@lbr38  Ok i try this week.

**Comment by jlebard** [comment]
désolé de répondre si tard, ça fait 15 jours que je suis sur un gros incident, donc à 100% dessus, je n'ai pas encore testé la fonction, je regarde ça, si tout ce passe bien, la semaine prochaine (sinon ça sera durant les vacances de noel, je dois déployer pour le 3 janvier ...)
